Framework/Core/src/LifetimeHelpers.cxx

```diff
@@ -275,6 +276,11 @@ ExpirationHandler::Handler LifetimeHelpers::enumerate(ConcreteDataMatcher const&
 {
   auto counter = std::make_shared<int64_t>(0);
   auto f = [matcher, counter, sourceChannel](ServiceRegistry& services, PartRef& ref, uint64_t timestamp) -> void {
+    // We generate entries according to our inputTimeslice.
+    if ((*counter % services.get<DeviceSpec>().maxInputTimeslices) != services.get<DeviceSpec>().inputTimesliceId) {
```

Comment:
There should be ``services.get<const DeviceSpec>`` here, I am getting:
``[ERROR] Unhandled exception reached the top of main: Unable to find service of kind N2o29framework10DeviceSpecE did you register one as non-const reference?, device shutting down.``